packages/cds-odata-annotation-converter/src/transforms/annotation/handlers/annotation.ts

```diff
@@ -161,8 +169,24 @@ function handleFlattenedStructure(state: VisitorState, annotation: Annotation, e
     // Build nested structures for CDS flattened syntax
     // e.g UI.Chart.AxisScaling.ScaleBehavior : #AutoScale, @Common.Text.@UI.TextArrangement : #TextFirst
     const flattenedSegments = getFlattenedSegments(state, annotation);
-
-    if (flattenedSegments.length) {
+    const { value } = annotation;
+    if (flattenedSegments?.[0]?.value === ReservedProperties.Type && value?.type === STRING_LITERAL_TYPE) {
+        const propertyRange = createRange(flattenedSegments[0].range?.start, value?.range?.end);
+        const recordWithType: Element = createElementNode({
+            name: Edm.Record,
+            range: propertyRange ?? undefined,
+            contentRange: propertyRange ?? undefined,
+            attributes: {
+                [Edm.Type]: createAttributeNode(Edm.Type, value.value, flattenedSegments[0].range, value.range)
+            }
+        });
+        element.content.push(recordWithType);
+        const currentState = state.popContext();
+        state.pushContext({
+            ...currentState,
+            isFlattened$Type: true
+        });
+    } else if (flattenedSegments.length) {
```

Comment:
This logic should be done in `convertFlattenedPath` function, because I suspect we would have a similar issue with cases where flattened syntax starts from a record property.

Reply:
adapted